src/main/java/jenkins/plugins/office365connector/Office365ConnectorWebhookNotifier.java

```diff
@@ -87,18 +88,14 @@ public static void sendBuildStaredNotification(Run run, TaskListener listener, b
 
         for (Webhook webhook : property.getWebhooks()) {
             if (webhook.isStartNotification()) {
-                //         listener.getLogger().println(String.format("Notifying webhook '%s'", webhook));
                 try {
                     HttpWorker worker = new HttpWorker(run.getEnvironment(listener).expand(webhook.getUrl()), gson.toJson(card), webhook.getTimeout(), 3, listener.getLogger());
                     executorService.submit(worker);
-                } catch (Throwable error) {
-                    error.printStackTrace(listener.error(String.format("Failed to notify webhook '%s'", webhook)));
-                    listener.getLogger().println(String.format("Failed to notify webhook '%s' - %s: %s", webhook, error.getClass().getName(), error.getMessage()));
+                } catch (IOException | InterruptedException | RejectedExecutionException e) {
```

Comment:
Throwable is not good idea. Also I recommend to switch https://codecov.io/ which works like Sonar but for pull requests.
I could not make it as I don't have permission for this project

Reply:
It's a good idea, but now, it won't make any difference as there is no UT's testing the card creation flows and code coverage would be abysmal. We can add this once we create UT's and bring code coverage to a respectable number.

Reply:
Agree. On the other hand this is not only about making difference but rather about code smell or clean code. I've already recommended code static analyzer so we can add some [badges](https://github.com/damianszczepanik/cucumber-reporting) or [comments](https://github.com/damianszczepanik/cucumber-reporting/pull/662) about the coverage but i don't have permission for this...

Never the less this is next step to make this code easier to maintain so I can add some new feature easier. I split changes to smaller so code review can be performed easier.